internal/store/hpa.go

```diff
@@ -255,6 +255,8 @@ var (
 						}
 					} else if intVal, canFastConvert := value.AsInt64(); canFastConvert {
 						metricValue = float64(intVal)
+					} else if c.Type == autoscaling.ExternalMetricSourceType {
```

Comment:
That would mean external metrics are always going to be specified in millis? Do we have docs for this?

Reply:
No, haven't found any docs yet, just made the assumption the same way we make the assumptions for other metrics. We could find if there is an `m` and only then do the calculation, otherwise leave the number as is?

Reply:
Are there other known representations of `resource.Quantity`? Other than plain intValues and `m`.

Reply:
Yes, there are many:

https://github.com/kubernetes/kube-state-metrics/blob/6b5f9ef06c61a94a27a93110fba7bc8ea84678e8/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L34-L48

Reply:
Update according to some k8s folks I asked, we should be using the ScalledValue, full msg:
> calling ScaledValue(0) - so first you get it unscaled, detect the order and see if scaling it will fit int64

Will give that a try.